modules/solid_mechanics/src/materials/CompositePowerLawCreepStressUpdate.C

```diff
@@ -0,0 +1,166 @@
+                     "non-integer m_exponent is used");
+}
+
+template <bool is_ad>
+void
+CompositePowerLawCreepStressUpdateTempl<is_ad>::computeStressInitialize(
+    const GenericReal<is_ad> & effective_trial_stress,
+    const GenericRankFourTensor<is_ad> & elasticity_tensor)
+{
+  RadialReturnStressUpdateTempl<is_ad>::computeStressInitialize(effective_trial_stress,
+                                                                elasticity_tensor);
+  _exp_time = std::pow(_t - _start_time, _m_exponent);
+}
+
+template <bool is_ad>
+template <typename ScalarType>
+ScalarType
+CompositePowerLawCreepStressUpdateTempl<is_ad>::computeResidualInternal(
+    const GenericReal<is_ad> & effective_trial_stress, const ScalarType & scalar)
+{
+  const ScalarType stress_delta = effective_trial_stress - _three_shear_modulus * scalar;
+  ScalarType creep_rate =
+      _coefficient[0] * std::pow(stress_delta, _n_exponent[0]) * _switchingFunc[0][_qp] *
+      std::exp(-_activation_energy[0] / (_gas_constant * (*_temperature)[_qp])) * _exp_time;
+  for (unsigned int n = 1; n < _num_materials; ++n)
+  {
+    creep_rate +=
+        _coefficient[n] * std::pow(stress_delta, _n_exponent[n]) * _switchingFunc[n][_qp] *
+        std::exp(-_activation_energy[n] / (_gas_constant * (*_temperature)[_qp])) * _exp_time;
+  }
+  return creep_rate * _dt - scalar;
+}
+
+template <bool is_ad>
+GenericReal<is_ad>
+CompositePowerLawCreepStressUpdateTempl<is_ad>::computeDerivative(
+    const GenericReal<is_ad> & effective_trial_stress, const GenericReal<is_ad> & scalar)
+{
+  const GenericReal<is_ad> stress_delta = effective_trial_stress - _three_shear_modulus * scalar;
+  GenericReal<is_ad> creep_rate_derivative =
+      -_coefficient[0] * _three_shear_modulus * _n_exponent[0] *
+      std::pow(stress_delta, _n_exponent[0] - 1.0) * _switchingFunc[0][_qp] *
+      std::exp(-_activation_energy[0] / (_gas_constant * (*_temperature)[_qp])) * _exp_time;
+
+  for (unsigned int n = 1; n < _num_materials; ++n)
+  {
+    creep_rate_derivative +=
+        -_coefficient[n] * _three_shear_modulus * _n_exponent[n] *
+        std::pow(stress_delta, _n_exponent[n] - 1.0) * _switchingFunc[n][_qp] *
+        std::exp(-_activation_energy[n] / (_gas_constant * (*_temperature)[_qp])) * _exp_time;
+  }
+  return creep_rate_derivative * _dt - 1.0;
+}
+
+template <bool is_ad>
+Real
+CompositePowerLawCreepStressUpdateTempl<is_ad>::computeStrainEnergyRateDensity(
+    const GenericMaterialProperty<RankTwoTensor, is_ad> & stress,
+    const GenericMaterialProperty<RankTwoTensor, is_ad> & strain_rate)
+{
+  GenericReal<is_ad> interpolated_exponent = 0.0;
+  for (unsigned int n = 0; n < _num_materials; ++n)
+  {
+    interpolated_exponent += (_n_exponent[n] / (_n_exponent[n] + 1)) * _switchingFunc[n][_qp];
+  }
+  return MetaPhysicL::raw_value(interpolated_exponent *
+                                stress[_qp].doubleContraction((strain_rate)[_qp]));
+}
+
```

Comment:
These lines of code aren't used in any of the regression tests. Design a regression test to call and test this method to increase the code coverage